### PR TITLE
Add xml of refrigerator alarm cluster and zap generated

### PR DESCRIPTION
**Pull request by jadhavrohit924**
- Changes:

> Add refrigerator alarm cluster xml to data-model.
> Zap generated for refrigerator alarm cluster.

- Testing:

> Build all-cluster-app and chip-tool

### CODE REVIEW

File: src/app/zap-templates/zcl/data-model/chip/refrigerator-alarm.xml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<configurator>
+  <domain name="CHIP"/>
+  <bitmap name="AlarmMap" type="BITMAP32">
+     <cluster code="0x0057" />
+     <item name="DoorOpen" value="0" mask="0x00"/>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/refrigerator-alarm.xml:21]:
```suggestion
     <item name="DoorOpen" mask="0x01"/>
```
at least that's what the spec seems to say....

File: src/app/zap-templates/zcl/data-model/chip/refrigerator-alarm.xml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    <attribute side="server" code="0x0001" define="LATCH" type="AlarmMap" default="0" writable="false" optional="false">Latch</attribute>
+    <attribute side="server" code="0x0002" define="STATE" type="AlarmMap" default="0" writable="false" optional="false">State</attribute>
+
+    <command source="client" code="00" name="Reset" optional="false">
+        <description>Reset alarm</description>
+        <arg name="Alarms" type="AlarmMap" optional="false"/>
+        <arg name="Mask" type="AlarmMap" optional="true"/>
+    </command>
+
+    <event side="server" code="0x00" priority="info" name="Notify" optional="false">
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/refrigerator-alarm.xml:41]: I don't see where the spec defines the event priority, but I guess OK for now.  Filed https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/7006

**Review by bzbarsky-apple — APPROVED** [review]
This looks like codegen did not get run properly...

**Comment by jadhavrohit924** [comment]
CI fails because of `IndentationError` in `src/controller/python/chip/clusters/Objects.py` On line `14815`. Which looks correct though. But everytime zap generates the same code with `IndentationError`. Am I missing something ?

File: src/app/zap-templates/zcl/data-model/chip/refrigerator-alarm.xml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<configurator>
+  <domain name="CHIP"/>
+  <bitmap name="AlarmMap" type="BITMAP32">
+     <cluster code="0x0057" />
+     <item name="DoorOpen" mask="0x01"/>
```
Comment (andy31415) [inline src/app/zap-templates/zcl/data-model/chip/refrigerator-alarm.xml:21]: Python codegen for this seems to have done:

```python

    class Bitmaps:
        class AlarmMap(IntFlag):
```

with empty content.

I believe the actual name for this should be `field` not `item` based on other examples

Comment (andy31415) [reply src/app/zap-templates/zcl/data-model/chip/refrigerator-alarm.xml:21]: @bzbarsky-apple @tecimovic is there a way to enforce xml validity? I believe we have a linter however we never enabled it. Would the linter have caught this?

Comment (bzbarsky-apple) [reply src/app/zap-templates/zcl/data-model/chip/refrigerator-alarm.xml:21]: I don't think we have a sane schema right now for this format...

Comment (jadhavrohit924) [reply src/app/zap-templates/zcl/data-model/chip/refrigerator-alarm.xml]: > I believe the actual name for this should be field not item based on other examples

Updated

**Comment by bzbarsky-apple** [comment]
I suspect the Darwin failures are from not enabling this client in controller-clusters.zap, because it turns out there _is_ one codegen dependency there.  https://github.com/project-chip/connectedhomeip/pull/26635 will fix that, though.

**Comment by bzbarsky-apple** [comment]
> https://github.com/project-chip/connectedhomeip/pull/26635 will fix that, though.

That merged, so rebased this on top of it and force-pushed.

**Review by dhrishi — APPROVED** [review]

File: src/app/zap-templates/zcl/data-model/chip/refrigerator-alarm.xml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+-->
+<configurator>
+  <domain name="CHIP"/>
+  <bitmap name="AlarmMap" type="BITMAP32">
+     <cluster code="0x0057" />
+     <field name="DoorOpen" mask="0x01"/>
+  </bitmap>
+  <cluster>
+    <name>Refrigerator Alarm</name>
+    <domain>Home &amp; Appliances</domain>
```
Comment (marktrayer) [inline src/app/zap-templates/zcl/data-model/chip/refrigerator-alarm.xml:25]: Please have the domain be just: Appliances.  
We need to have this consistent across all clusters from the Appliance Team in 1.2.  
Thanks!

Comment (jadhavrohit924) [reply src/app/zap-templates/zcl/data-model/chip/refrigerator-alarm.xml]: Done

**Review by andy31415 — APPROVED** [review]

**Pull request merged**